Userland/Utilities/errno.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <stdlib.h>
+
+int main(int argc, char** argv)
+{
+    bool list = false;
+    bool search = false;
+    const char* keyword = nullptr;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(keyword, "Error number or string to search", "keyword", Core::ArgsParser::Required::No);
+    args_parser.add_option(list, "List all errno values", "list", 'l');
+    args_parser.add_option(search, "Search for error descriptions containing keyword", "search", 's');
+    args_parser.parse(argc, argv);
+
+    if (list) {
+        for (int i = 0; i < sys_nerr; i++) {
+            outln("{} {}", i, strerror(i));
+        }
+        return 0;
+    }
+
+    if (keyword == nullptr)
+        return 0;
+
+    if (search) {
+        for (int i = 0; i < sys_nerr; i++) {
+            auto error = String::formatted("{}", strerror(i));
+            if (error.contains(keyword, CaseSensitivity::CaseInsensitive)) {
+                outln("{} {}", i, error);
+            }
+        }
+        return 0;
+    }
+
+    auto error = String::formatted("{}", strerror(atoi(keyword)));
+    if (error == "Unknown error")
+        return 1;
```

Comment:
Maybe we should print an error message for that case?

Reply:
That depends on how close we want to follow existing convention. No error is printed on Linux:

```
user@ubuntu:~/Desktop/serenity$ errno 12345
user@ubuntu:~/Desktop/serenity$ 
```

Reply:
Fair enough. I don't have this tool installed on my machine, but feel free to copy its behaviour for the comment above as well.

Reply:
Done. Printed an error in both instances.